src/corefx/io/corefx_io.cpp

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include "corefx_io.h"
+#include <sys/stat.h>
+
+#if HAVE_STAT64 && !(defined(__APPLE__) && defined(_AMD64_))
+#    define stat_ stat64
+#    define fstat_ fstat64
+#else
+#   define stat_ stat
+#   define fstat_ fstat
+#endif
+
+static void ConvertFileStats(const struct stat_& src, FileStats* dst)
+{
+    dst->Flags = FILESTATS_FLAGS_NONE;
+    dst->Mode = src.st_mode;
```

Comment:
Are there any systems that use different values for the named flags that make up the st_mode?  I seem to remember @ellismg verifying "no", at least for the systems we were currently targeting, but I don't know that that's always going to be true, in which case we'd need to do conversions for fields like this.